Onepassword

snarlysodboxer opened the pull request:
Support 1Password as a provider.

snarlysodboxer committed:
Add 1Password support

EdgeJ commented on docs/provider-1password-automation.md:
```diff
@@ -0,0 +1,29 @@
+# TODO incomplete notes about documentation to write
+The 1Password provider supports secrets types of `Password` and `Document`.
+* The `Password` type can get data from multiple `fields` in the entry, but not the actual password field or anything else. See [creating 1Password secrets compatible with external-secrets](#creating-compatible-secrets).
+* The `Document` type can get data from the main file (not the actual password field).
```
I find this a little confusing.  Neither type can fetch password fields?

snarlysodboxer replied:
oh my, don't read these yet, these we're just notes to myself, i will fix up the documentation and then ask for review on it. :)

snarlysodboxer replied:
Docs are now updated.

EdgeJ commented on pkg/provider/onepassword/onepassword_test.go:
```diff
@@ -0,0 +1,362 @@
+/*
```
Tests look good.  Only minor (and optional) nit I have is they might be easier to read if they were re-written with a table-driven style.

snarlysodboxer replied:
re-written

snarlysodboxer replied:
Finished the rest. Everything is table driven now.

EdgeJ reviewed (approved):
One suggestion and one question about some documentation.  Otherwise, LGTM

snarlysodboxer commented on apis/externalsecrets/v1alpha1/secretstore_onepassword_types.go:
```diff
@@ -31,7 +31,10 @@ type OnePasswordAuthSecretRef struct {
+	// ConnectHost defines the OnePassword Connect Server to connect to
+	ConnectHost string `json:"connectHost"`
+	// Vaults defines which OnePassword vaults to search in which order
+	Vaults map[string]int `json:"vaults"`
```
These comments make their way in to the generated CRDs and help kubectl explain work better. (A common practice in operators, not just this one.)

dfsdevops commented on apis/externalsecrets/v1alpha1/zz_generated.deepcopy.go:
```diff
@@ -775,8 +775,10 @@ func (in *OnePasswordProvider) DeepCopyInto(out *OnePasswordProvider) {
 		in, out := &in.Vaults, &out.Vaults
-		*out = make([]string, len(*in))
-		copy(*out, *in)
+		*out = make(map[string]int, len(*in))
```
I never provided a length to a make(map) before,  guessing it helps optimize

dfsdevops replied:
ooh this is autogenerated... doh!

dfsdevops commented on apis/externalsecrets/v1alpha1/zz_generated.deepcopy.go:
```diff
@@ -775,8 +775,10 @@ func (in *OnePasswordProvider) DeepCopyInto(out *OnePasswordProvider) {
-		copy(*out, *in)
+		*out = make(map[string]int, len(*in))
+		for key, val := range *in {
+			(*out)[key] = val
```
So funny that go doesn't have any built-in helper for copying maps.

dfsdevops commented on pkg/provider/onepassword/fake/fake.go:
```diff
@@ -67,39 +68,44 @@ func (mockClient *OnePasswordMockClient) GetFileContent(file *onepassword.File)
-func (mockClient *OnePasswordMockClient) AddItemFields(itemFields map[string][]*onepassword.ItemField) *OnePasswordMockClient {
-	for key, value := range itemFields {
-		mockClient.MockItemFields[key] = value
+func (mockClient *OnePasswordMockClient) AddItemFields(itemFieldsMap map[string]map[string][]*onepassword.ItemField) *OnePasswordMockClient {
```
```go
map[string]map[string][]*onepassword.ItemField
```
seems like kind of a mouthful, is there some way to simplify this by using a custom type or similar? Not a big deal it just makes my eyes glaze over.

snarlysodboxer replied:
went over this in slack and and changed it for the better!

dfsdevops commented on pkg/provider/onepassword/onepassword.go:
```diff
@@ -197,6 +199,33 @@ func countFieldsWithLabel(fieldLabel string, fields []*onepassword.ItemField) in
+func (list orderedVaultList) Swap(i, j int)      { list[i], list[j] = list[j], list[i] }
+func (list orderedVaultList) Less(i, j int) bool { return list[i].Order < list[j].Order }
+
+func sortVaults(vaults *map[string]int) []string {
```
I find it a little confusing when pointers to maps are used as function inputs (unless it's one of the few cases where you really need it to be), since they are already a reference type so it should be a cheap copy, and we aren't mutating it (though I think you can still mutate a map when it's passed without a pointer..)

I was also reading earlier that using pointers can actually make go programs slower in some cases because it causes the GC to work harder to figure out if the reference needs to be cleaned off the heap, whereas if you use a normal input it just adds it to the stack and pops it off when it returns.

I'm still in my go learning process so I could be way off here.... probably just a matter of taste in most circumstances. Also, not a blocker, please don't hate me for this review I'm just using it to learn and validate my understanding 🤣

snarlysodboxer replied:
You're right! Thanks for asking about that. I have removed all pointers to maps. :)

gusfcarvalho committed:
WIP: GetAllSecrets for vault method

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
Adding GetAllSecrets for Hashicorp Vault

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

kinyat committed:
Add the ability to support scoped RBAC with a scoped namespace

paul-the-alien[bot] committed:
Merge pull request #807 from lianmakesthings/improve-gcp-docs

Improve GCP docs

AtzeDeVries committed:
use storespec directly in get ClusterProjectID

zen committed:
Added note about ParameterStore 

This snippet is too generic and it may confuse users of Parameter Store backend in provided examples .

paul-the-alien[bot] committed:
Merge pull request #811 from zen/patch-1

Added note about ParameterStore

AtzeDeVries committed:
fix lint

gusfcarvalho committed:
Adding docs and implementing ConversionStrategy

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #783 from AtzeDeVries/allow-gcp-cross-project-secrets

GCP: allow cluster to be in different project

gusfcarvalho committed:
Changes e2e test default api to v1beta1. Adds GetAllSecrets test

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
Moving duplication validation logic to utils method

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
Fixing lint

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
Adding missing changes from v1alpha1 and v1beta1

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

Commit:
Adding warning session to docs

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
Adding EngineVersion=v1 to GCP e2e tests

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
Chore: bumping gitlab-go to 0.58.0

Signed-off-by: Gustavo Carvalho <gusfcarvalho@gmail.com>

paul-the-alien[bot] committed:
Merge pull request #814 from external-secrets/chore/update-gitlab-go

Chore: bumping gitlab-go to 0.58.0

haf-tech committed:
Minor change - refreshInterval 60m

this template is used mostly as is, and the previous `refreshInterval` of `1m` is to high and produces only too much pressure on the API endpoints

kinyat committed:
skip processing cluster secret store

Commit:
Add flags to control weather a deployment should be created

paul-the-alien[bot] committed:
Merge pull request #817 from kinyat/feat/control-enables

Add `create` flags to control weather deployments should be created

paul-the-alien[bot] committed:
Merge pull request #815 from haf-tech/patch-1

Minor change - IBM Cloud SM example - refreshInterval 60m

Commit:
Add kubernetes provider documentation

gusfcarvalho committed:
Updating PAUL.yaml to allow 30 PRs (#819)

Signed-off-by: Gustavo Carvalho <gusfcarvalho@gmail.com>

Co-authored-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #784 from external-secrets/feature/hashivault-getallsecrets

Implements Hashicorp Vault GetAllSecrets

kinyat committed:
make scopedRBAC to disable cluster secret store

kinyat committed:
refactor and add unit test for skipping cluster secret store

dependabot[bot] committed:
build(deps): bump github.com/golang-jwt/jwt/v4 from 4.2.0 to 4.3.0

Bumps [github.com/golang-jwt/jwt/v4](https://github.com/golang-jwt/jwt) from 4.2.0 to 4.3.0.
- [Release notes](https://github.com/golang-jwt/jwt/releases)
- [Changelog](https://github.com/golang-jwt/jwt/blob/main/VERSION_HISTORY.md)
- [Commits](https://github.com/golang-jwt/jwt/compare/v4.2.0...v4.3.0)

---
updated-dependencies:
- dependency-name: github.com/golang-jwt/jwt/v4
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/Azure/azure-sdk-for-go

Bumps [github.com/Azure/azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go) from 62.1.0+incompatible to 62.2.0+incompatible.
- [Release notes](https://github.com/Azure/azure-sdk-for-go/releases)
- [Changelog](https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/Azure/azure-sdk-for-go/compare/v62.1.0...v62.2.0)

---
updated-dependencies:
- dependency-name: github.com/Azure/azure-sdk-for-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/spf13/cobra from 1.3.0 to 1.4.0

Bumps [github.com/spf13/cobra](https://github.com/spf13/cobra) from 1.3.0 to 1.4.0.
- [Release notes](https://github.com/spf13/cobra/releases)
- [Changelog](https://github.com/spf13/cobra/blob/master/CHANGELOG.md)
- [Commits](https://github.com/spf13/cobra/compare/v1.3.0...v1.4.0)

---
updated-dependencies:
- dependency-name: github.com/spf13/cobra
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
Add cluster secrets and update secretStore examples

Commit:
updates spec

Commit:
add k8s to supported backends

paul-the-alien[bot] committed:
Merge pull request #826 from rodrmartinez/docs/kubernetes-provider

Docs/kubernetes provider

paul-the-alien[bot] committed:
Merge pull request #825 from external-secrets/dependabot/go_modules/github.com/spf13/cobra-1.4.0

build(deps): bump github.com/spf13/cobra from 1.3.0 to 1.4.0

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.44.0 to 1.45.0

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.44.0 to 1.45.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.44.0...v1.45.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

paul-the-alien[bot] committed:
Merge pull request #824 from external-secrets/dependabot/go_modules/github.com/Azure/azure-sdk-for-go-62.2.0incompatible

build(deps): bump github.com/Azure/azure-sdk-for-go from 62.1.0+incompatible to 62.2.0+incompatible

paul-the-alien[bot] committed:
Merge pull request #821 from external-secrets/dependabot/go_modules/google.golang.org/grpc-1.45.0

build(deps): bump google.golang.org/grpc from 1.44.0 to 1.45.0

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.70.0 to 0.71.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.70.0 to 0.71.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.70.0...v0.71.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

paul-the-alien[bot] committed:
Merge pull request #822 from external-secrets/dependabot/go_modules/github.com/golang-jwt/jwt/v4-4.3.0

build(deps): bump github.com/golang-jwt/jwt/v4 from 4.2.0 to 4.3.0

paul-the-alien[bot] committed:
Merge pull request #823 from external-secrets/dependabot/go_modules/google.golang.org/api-0.71.0

build(deps): bump google.golang.org/api from 0.70.0 to 0.71.0

AndreyZamyslov committed:
Yandex Lockbox Connection Leaks (#835)

* fix Yandex Lockbox provider connection leaks (https://github.com/external-secrets/external-secrets/issues/833)

* fix Yandex Lockbox provider connection leaks (https://github.com/external-secrets/external-secrets/issues/833)

gusfcarvalho committed:
Added draft proposal for Secret Sink (#641)

* Added draft for SecretSink

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

* Fixing typo and adding more comments

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

* Dedicating new CRD to SecretSink

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

* Redoing secretSink proposal

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

* Moved selector to spec level. Removed secretStore selector from remoteRef

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

* Update wording on design/001-secretsink.md

Co-authored-by: Moritz Johner <moolen@users.noreply.github.com>
Co-authored-by: Moritz Johner <moolen@users.noreply.github.com>

moolen committed:
chore: update yandex sdk

moolen committed:
fix: integrate sdk changes

paul-the-alien[bot] committed:
Merge pull request #844 from external-secrets/update-yandex-sdk

chore: update yandex sdk

cpanato committed:
sign images using cosign (#845)

Signed-off-by: cpanato <ctadeu@gmail.com>

ADustyOldMuffin committed:
feat: implement ClusterExternalSecret (#542)

Co-authored-by: Gustavo Fernandes de Carvalho <gusfcarvalho@gmail.com>

Commit:
fix: update CRDs

nitzann committed:
Add public_cert support

cpanato committed:
add missing install of cosign in the CIs pipelines

Signed-off-by: cpanato <ctadeu@gmail.com>

paul-the-alien[bot] committed:
Merge pull request #846 from external-secrets/fix/crd-gen

fix: update CRDs

nitzann committed:
Add public_cert support

knelasevero committed:
Merge pull request #848 from cpanato/follow-731

add missing install of cosign in the CIs pipelines

dependabot[bot] committed:
build(deps): bump helm/chart-testing-action from 2.2.0 to 2.2.1

Bumps [helm/chart-testing-action](https://github.com/helm/chart-testing-action) from 2.2.0 to 2.2.1.
- [Release notes](https://github.com/helm/chart-testing-action/releases)
- [Commits](https://github.com/helm/chart-testing-action/compare/v2.2.0...v2.2.1)

---
updated-dependencies:
- dependency-name: helm/chart-testing-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump helm/chart-testing-action from 2.2.0 to 2.2.1

Bumps [helm/chart-testing-action](https://github.com/helm/chart-testing-action) from 2.2.0 to 2.2.1.
- [Release notes](https://github.com/helm/chart-testing-action/releases)
- [Commits](https://github.com/helm/chart-testing-action/compare/v2.2.0...v2.2.1)

---
updated-dependencies:
- dependency-name: helm/chart-testing-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/googleapis/gax-go/v2 from 2.1.1 to 2.2.0

Bumps [github.com/googleapis/gax-go/v2](https://github.com/googleapis/gax-go) from 2.1.1 to 2.2.0.
- [Release notes](https://github.com/googleapis/gax-go/releases)
- [Commits](https://github.com/googleapis/gax-go/compare/v2.1.1...v2.2.0)

---
updated-dependencies:
- dependency-name: github.com/googleapis/gax-go/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/xanzy/go-gitlab from 0.58.0 to 0.60.0

Bumps [github.com/xanzy/go-gitlab](https://github.com/xanzy/go-gitlab) from 0.58.0 to 0.60.0.
- [Release notes](https://github.com/xanzy/go-gitlab/releases)
- [Changelog](https://github.com/xanzy/go-gitlab/blob/master/releases_test.go)
- [Commits](https://github.com/xanzy/go-gitlab/compare/v0.58.0...v0.60.0)

---
updated-dependencies:
- dependency-name: github.com/xanzy/go-gitlab
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/IBM/go-sdk-core/v5 from 5.9.2 to 5.9.3

Bumps [github.com/IBM/go-sdk-core/v5](https://github.com/IBM/go-sdk-core) from 5.9.2 to 5.9.3.
- [Release notes](https://github.com/IBM/go-sdk-core/releases)
- [Changelog](https://github.com/IBM/go-sdk-core/blob/main/CHANGELOG.md)
- [Commits](https://github.com/IBM/go-sdk-core/compare/v5.9.2...v5.9.3)

---
updated-dependencies:
- dependency-name: github.com/IBM/go-sdk-core/v5
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

paul-the-alien[bot] committed:
Merge pull request #856 from external-secrets/dependabot/go_modules/github.com/IBM/go-sdk-core/v5-5.9.3

build(deps): bump github.com/IBM/go-sdk-core/v5 from 5.9.2 to 5.9.3

dependabot[bot] committed:
build(deps): bump github.com/IBM/secrets-manager-go-sdk

Bumps [github.com/IBM/secrets-manager-go-sdk](https://github.com/IBM/secrets-manager-go-sdk) from 1.0.36 to 1.0.37.
- [Release notes](https://github.com/IBM/secrets-manager-go-sdk/releases)
- [Commits](https://github.com/IBM/secrets-manager-go-sdk/compare/v1.0.36...v1.0.37)

---
updated-dependencies:
- dependency-name: github.com/IBM/secrets-manager-go-sdk
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

paul-the-alien[bot] committed:
Merge pull request #853 from external-secrets/dependabot/go_modules/github.com/googleapis/gax-go/v2-2.2.0

build(deps): bump github.com/googleapis/gax-go/v2 from 2.1.1 to 2.2.0

paul-the-alien[bot] committed:
Merge pull request #855 from external-secrets/dependabot/go_modules/github.com/xanzy/go-gitlab-0.60.0

build(deps): bump github.com/xanzy/go-gitlab from 0.58.0 to 0.60.0

dependabot[bot] committed:
build(deps): bump github.com/stretchr/testify from 1.7.0 to 1.7.1

Bumps [github.com/stretchr/testify](https://github.com/stretchr/testify) from 1.7.0 to 1.7.1.
- [Release notes](https://github.com/stretchr/testify/releases)
- [Commits](https://github.com/stretchr/testify/compare/v1.7.0...v1.7.1)

---
updated-dependencies:
- dependency-name: github.com/stretchr/testify
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

paul-the-alien[bot] committed:
Merge pull request #854 from external-secrets/dependabot/go_modules/github.com/IBM/secrets-manager-go-sdk-1.0.37

build(deps): bump github.com/IBM/secrets-manager-go-sdk from 1.0.36 to 1.0.37

gusfcarvalho committed:
Adding ValidateStore for fake provider

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
Adding ValidateStore for IBM provider. Improving util check

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #852 from external-secrets/dependabot/go_modules/github.com/stretchr/testify-1.7.1

build(deps): bump github.com/stretchr/testify from 1.7.0 to 1.7.1

haf-tech committed:
Enhance IBM Secrets Manager support with kv secretType

paul-the-alien[bot] committed:
Merge pull request #851 from external-secrets/dependabot/github_actions/helm/chart-testing-action-2.2.1

build(deps): bump helm/chart-testing-action from 2.2.0 to 2.2.1

haf-tech committed:
Remove unused ref

haf-tech committed:
After make fmt

nitzann committed:
Update docs/provider-ibm-secrets-manager.md

Co-authored-by: Lucas Severo Alves <lucassalves65@gmail.com>

haf-tech committed:
Finalize kv secretType support for IBM Cloud SM

haf-tech committed:
Finalize kv secretType support for IBM Cloud SM. Fix fmt

gusfcarvalho committed:
Adding mutexes only for GCP provider

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

pandeybk committed:
How to manage secrets in OpenShift using Vault and External Secrets Operator blog reference

Signed-off-by: Balkrishna Pandey <sachit.nep@gmail.com>

haf-tech committed:
Optimize lint/fmt issues

haf-tech committed:
Optimize lint/fmt issues

haf-tech committed:
Optimize lint/fmt issues

haf-tech committed:
Optimize lint/fmt issues

haf-tech committed:
Optimize lint/fmt issues

moolen committed:
feat(azure): implement workload identity (#738)

* feat(azure): implement workload identity

Signed-off-by: Moritz Johner <beller.moritz@googlemail.com>
Co-authored-by: Henning Eggers <henning.eggers@inovex.de>

paul-the-alien[bot] committed:
Merge pull request #874 from pandeybk/main

How to manage secrets in OpenShift using Vault and External Secrets Operator blog reference

Commit:
add Balkrishna Pandey demo

paul-the-alien[bot] committed:
Merge pull request #875 from external-secrets/add-demo

Add Balkrishna Pandey demo

haf-tech committed:
Optimize lint/fmt issues

gusfcarvalho committed:
Adding docs for v1beta1 vs v1alpha1. Added one test for v1alpha1 compatibility

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
Fixing setup-gcloud version

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #878 from external-secrets/fix/setup-gcloud-version

Fixing setup-gcloud version

gusfcarvalho committed:
Setting version to v0 as main is not yet available

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #879 from external-secrets/fix/setup-gcloud-version

Setting version to v0 as main is not yet available

gusfcarvalho committed:
Removing Key checks from utils.go and passing them to IBM provider

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
Marking v1alpha1 as deprecated.

Improving docs and menu order.
Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #849 from ibm-cloud-security/main

Add public_cert support

haf-tech committed:
Merge from main

paul-the-alien[bot] committed:
Merge pull request #868 from external-secrets/chore/validate-store-for-providers

Adds ValidateStore for some providers

paul-the-alien[bot] committed:
Merge pull request #873 from gusfcarvalho/fix/mutex-on-gcp

Adding mutexes only for GCP provider

kozaktomas committed:
Fix invalid json in documentation

paul-the-alien[bot] committed:
Merge pull request #884 from kozaktomas/fix-docs-json-2

Fix invalid json in documentation

kinyat committed:
add new flags to control both cluster store and cluster external secret reconciler

moolen committed:
feature: aws getallsecrets (#820)

* feature: aws getallsecrets

Signed-off-by: Moritz Johner <beller.moritz@googlemail.com>

* fix: e2e test / find by name

* feat: add get-by-tags tests, consolidate with existing ones

Signed-off-by: Moritz Johner <beller.moritz@googlemail.com>

* feat: add path tests

* fix: revert azure

* fix: secretsmanager prealloc

* feat: aws sm path tests

* feat: implement secretsmanager path filter

* fix: use low refresh interval due to eventual consistency

* revert makefile changes

* fix: add path test cases to managed

Co-authored-by: Docs <docs@external-secrets.io>

haf-tech committed:
Enhance logic support . in key and returning entire payload if no property is set

haf-tech committed:
Fix lint issues

haf-tech committed:
Fix lint/fmt issues

Commit:
Add namespace to service resource & fix typo

Signed-off-by: Suraj Narwade <suraj.narwade@rvu.co.uk>

burak-yuksel committed:
Validate for Kubernetes Provider

burak-yuksel committed:
For failing tests

burak-yuksel committed:
For failing SonarCloud tests

gusfcarvalho committed:
Removing namespace references from full-secret-store.yaml

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #895 from surajnarwade/fix-docs

Add namespace to service resource & fix typo

paul-the-alien[bot] committed:
Merge pull request #898 from external-secrets/fix/full-secret-store-snippet

Removing namespace references from full-secret-store.yaml

paul-the-alien[bot] committed:
Merge pull request #800 from kinyat/feat/scoped-rbac

Add the ability to support scoped RBAC with a scoped namespace

Commit:
improves kubernetes provider documentation

Commit:
implements ValidateStore method

Commit:
Fixed some masked error and avoided panics

Commit:
Merge from main

Commit:
Fixed lint issue

Commit:
Add Vault 1.10 updates

haf-tech committed:
Enhance IBM SM provider for SecretMap

haf-tech committed:
Fix lint/fmt issues

dependabot[bot] committed:
build(deps): bump actions/cache from 2.1.7 to 3

Bumps [actions/cache](https://github.com/actions/cache) from 2.1.7 to 3.
- [Release notes](https://github.com/actions/cache/releases)
- [Commits](https://github.com/actions/cache/compare/v2.1.7...v3)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump peter-evans/slash-command-dispatch from 2.3.0 to 3

Bumps [peter-evans/slash-command-dispatch](https://github.com/peter-evans/slash-command-dispatch) from 2.3.0 to 3.
- [Release notes](https://github.com/peter-evans/slash-command-dispatch/releases)
- [Commits](https://github.com/peter-evans/slash-command-dispatch/compare/v2.3.0...v3)

---
updated-dependencies:
- dependency-name: peter-evans/slash-command-dispatch
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/onsi/gomega from 1.18.1 to 1.19.0

Bumps [github.com/onsi/gomega](https://github.com/onsi/gomega) from 1.18.1 to 1.19.0.
- [Release notes](https://github.com/onsi/gomega/releases)
- [Changelog](https://github.com/onsi/gomega/blob/master/CHANGELOG.md)
- [Commits](https://github.com/onsi/gomega/compare/v1.18.1...v1.19.0)

---
updated-dependencies:
- dependency-name: github.com/onsi/gomega
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/go-logr/logr from 1.2.2 to 1.2.3

Bumps [github.com/go-logr/logr](https://github.com/go-logr/logr) from 1.2.2 to 1.2.3.
- [Release notes](https://github.com/go-logr/logr/releases)
- [Changelog](https://github.com/go-logr/logr/blob/master/CHANGELOG.md)
- [Commits](https://github.com/go-logr/logr/compare/v1.2.2...v1.2.3)

---
updated-dependencies:
- dependency-name: github.com/go-logr/logr
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/akeylesslabs/akeyless-go/v2

Bumps [github.com/akeylesslabs/akeyless-go/v2](https://github.com/akeylesslabs/akeyless-go) from 2.15.25 to 2.15.28.
- [Release notes](https://github.com/akeylesslabs/akeyless-go/releases)
- [Changelog](https://github.com/akeylesslabs/akeyless-go/blob/master/docs/KmipRenewServerCertificate.md)
- [Commits](https://github.com/akeylesslabs/akeyless-go/compare/v2.15.25...v2.15.28)

---
updated-dependencies:
- dependency-name: github.com/akeylesslabs/akeyless-go/v2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.71.0 to 0.73.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.71.0 to 0.73.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.71.0...v0.73.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/Azure/azure-sdk-for-go

Bumps [github.com/Azure/azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go) from 62.2.0+incompatible to 62.3.0+incompatible.
- [Release notes](https://github.com/Azure/azure-sdk-for-go/releases)
- [Changelog](https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/Azure/azure-sdk-for-go/compare/v62.2.0...v62.3.0)

---
updated-dependencies:
- dependency-name: github.com/Azure/azure-sdk-for-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

kinyat committed:
add create flags to each CRDs resources for a more fine-grained control

paul-the-alien[bot] committed:
Merge pull request #902 from external-secrets/dependabot/github_actions/actions/cache-3

build(deps): bump actions/cache from 2.1.7 to 3

paul-the-alien[bot] committed:
Merge pull request #908 from external-secrets/dependabot/go_modules/github.com/Azure/azure-sdk-for-go-62.3.0incompatible

build(deps): bump github.com/Azure/azure-sdk-for-go from 62.2.0+incompatible to 62.3.0+incompatible

paul-the-alien[bot] committed:
Merge pull request #907 from external-secrets/dependabot/go_modules/google.golang.org/api-0.73.0

build(deps): bump google.golang.org/api from 0.71.0 to 0.73.0

paul-the-alien[bot] committed:
Merge pull request #906 from external-secrets/dependabot/go_modules/github.com/akeylesslabs/akeyless-go/v2-2.15.28

build(deps): bump github.com/akeylesslabs/akeyless-go/v2 from 2.15.25 to 2.15.28

paul-the-alien[bot] committed:
Merge pull request #905 from external-secrets/dependabot/go_modules/github.com/go-logr/logr-1.2.3

build(deps): bump github.com/go-logr/logr from 1.2.2 to 1.2.3

paul-the-alien[bot] committed:
Merge pull request #904 from external-secrets/dependabot/go_modules/github.com/onsi/gomega-1.19.0

build(deps): bump github.com/onsi/gomega from 1.18.1 to 1.19.0

paul-the-alien[bot] committed:
Merge pull request #903 from external-secrets/dependabot/github_actions/peter-evans/slash-command-dispatch-3

build(deps): bump peter-evans/slash-command-dispatch from 2.3.0 to 3

Commit:
replace remote server url

paul-the-alien[bot] committed:
Merge pull request #899 from rodrmartinez/chore/kubernetes-provider-docs-update

Improves kubernetes provider examples documentation

LukeMwila committed:
docs: added link to video on using eso with aws secrets manager

Signed-off-by: Lukonde Mwila <outlier.developer@gmail.com>

kinyat committed:
Only generate create flags for cluster scoped crds

paul-the-alien[bot] committed:
Merge pull request #901 from jbayer/vault-110-update

Add Vault 1.10 updates

LukeMwila committed:
Update docs/eso-demos.md

Co-authored-by: Lucas Severo Alves <lucassalves65@gmail.com>

paul-the-alien[bot] committed:
Merge pull request #880 from external-secrets/chore/docs-improvements

Chore/docs improvements

Commit:
Adds ValidateStore testing

burak-yuksel committed:
Merge remote-tracking branch 'origin/main' into feature/validate-kubernetes-provider

burak-yuksel committed:
Documentation note added for required permission

paul-the-alien[bot] committed:
Merge pull request #911 from LukeMwila/docs/lmwila/add-to-eso-demos

docs: added link to video on using eso with aws secrets manager

haf-tech committed:
Enhance and optimize IBM Provider docu

Commit:
Adds ValidateSecretSelector

paul-the-alien[bot] committed:
Merge pull request #805 from external-secrets/feature/azure-getallsecrets

Implement GetAllSecrets for Azure Key Vault

paul-the-alien[bot] committed:
Merge pull request #910 from kinyat/feat/crds-create-flags

add create flags to each CRDs resources for a more fine-grained control

Commit:
fix lint errors

gusfcarvalho committed:
Merge pull request #870 from haf-tech/ibmcloud-sm-kv2

Enhance IBM Secrets Manager support with kv secretType

paul-the-alien[bot] committed:
Merge pull request #896 from burak-yuksel/feature/validate-kubernetes-provider

Validate for Kubernetes Provider

dependabot[bot] committed:
build(deps): bump helm/chart-releaser-action from 1.3.0 to 1.4.0

Bumps [helm/chart-releaser-action](https://github.com/helm/chart-releaser-action) from 1.3.0 to 1.4.0.
- [Release notes](https://github.com/helm/chart-releaser-action/releases)
- [Commits](https://github.com/helm/chart-releaser-action/compare/v1.3.0...v1.4.0)

---
updated-dependencies:
- dependency-name: helm/chart-releaser-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump actions/setup-python from 2 to 3.1.0

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 2 to 3.1.0.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v2...v3.1.0)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump terraform-linters/setup-tflint from 1 to 2

Bumps [terraform-linters/setup-tflint](https://github.com/terraform-linters/setup-tflint) from 1 to 2.
- [Release notes](https://github.com/terraform-linters/setup-tflint/releases)
- [Commits](https://github.com/terraform-linters/setup-tflint/compare/v1...v2)

---
updated-dependencies:
- dependency-name: terraform-linters/setup-tflint
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump sigs.k8s.io/controller-runtime from 0.11.1 to 0.11.2

Bumps [sigs.k8s.io/controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) from 0.11.1 to 0.11.2.
- [Release notes](https://github.com/kubernetes-sigs/controller-runtime/releases)
- [Commits](https://github.com/kubernetes-sigs/controller-runtime/compare/v0.11.1...v0.11.2)

---
updated-dependencies:
- dependency-name: sigs.k8s.io/controller-runtime
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/api from 0.73.0 to 0.74.0

Bumps [google.golang.org/api](https://github.com/googleapis/google-api-go-client) from 0.73.0 to 0.74.0.
- [Release notes](https://github.com/googleapis/google-api-go-client/releases)
- [Changelog](https://github.com/googleapis/google-api-go-client/blob/main/CHANGES.md)
- [Commits](https://github.com/googleapis/google-api-go-client/compare/v0.73.0...v0.74.0)

---
updated-dependencies:
- dependency-name: google.golang.org/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/akeylesslabs/akeyless-go/v2

Bumps [github.com/akeylesslabs/akeyless-go/v2](https://github.com/akeylesslabs/akeyless-go) from 2.15.28 to 2.15.29.
- [Release notes](https://github.com/akeylesslabs/akeyless-go/releases)
- [Changelog](https://github.com/akeylesslabs/akeyless-go/blob/master/docs/KmipRenewServerCertificate.md)
- [Commits](https://github.com/akeylesslabs/akeyless-go/compare/v2.15.28...v2.15.29)

---
updated-dependencies:
- dependency-name: github.com/akeylesslabs/akeyless-go/v2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/xanzy/go-gitlab from 0.60.0 to 0.61.0

Bumps [github.com/xanzy/go-gitlab](https://github.com/xanzy/go-gitlab) from 0.60.0 to 0.61.0.
- [Release notes](https://github.com/xanzy/go-gitlab/releases)
- [Changelog](https://github.com/xanzy/go-gitlab/blob/master/releases_test.go)
- [Commits](https://github.com/xanzy/go-gitlab/compare/v0.60.0...v0.61.0)

---
updated-dependencies:
- dependency-name: github.com/xanzy/go-gitlab
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

merlindorin committed:
fix: use controller-runtime internal backoff retry instead of explicitly ask a fixed retry

paul-the-alien[bot] committed:
Merge pull request #926 from external-secrets/dependabot/github_actions/helm/chart-releaser-action-1.4.0

build(deps): bump helm/chart-releaser-action from 1.3.0 to 1.4.0

gusfcarvalho committed:
GetAllSecrets docs

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #927 from external-secrets/dependabot/github_actions/actions/setup-python-3.1.0

build(deps): bump actions/setup-python from 2 to 3.1.0

paul-the-alien[bot] committed:
Merge pull request #928 from external-secrets/dependabot/github_actions/terraform-linters/setup-tflint-2

build(deps): bump terraform-linters/setup-tflint from 1 to 2

paul-the-alien[bot] committed:
Merge pull request #929 from external-secrets/dependabot/go_modules/sigs.k8s.io/controller-runtime-0.11.2

build(deps): bump sigs.k8s.io/controller-runtime from 0.11.1 to 0.11.2

paul-the-alien[bot] committed:
Merge pull request #930 from external-secrets/dependabot/go_modules/google.golang.org/api-0.74.0

build(deps): bump google.golang.org/api from 0.73.0 to 0.74.0

paul-the-alien[bot] committed:
Merge pull request #931 from external-secrets/dependabot/go_modules/github.com/akeylesslabs/akeyless-go/v2-2.15.29

build(deps): bump github.com/akeylesslabs/akeyless-go/v2 from 2.15.28 to 2.15.29

paul-the-alien[bot] committed:
Merge pull request #933 from external-secrets/dependabot/go_modules/github.com/xanzy/go-gitlab-0.61.0

build(deps): bump github.com/xanzy/go-gitlab from 0.60.0 to 0.61.0

alfredkrohmer committed:
feat(vault): add option for JWT backend to authenticate with Kubernetes service account token (#768)

gusfcarvalho committed:
Update docs/guides-getallsecrets.md

Co-authored-by: Moritz Johner <moolen@users.noreply.github.com>

gusfcarvalho committed:
Update docs/guides-getallsecrets.md

Co-authored-by: Moritz Johner <moolen@users.noreply.github.com>

gusfcarvalho committed:
Update docs/guides-getallsecrets.md

Co-authored-by: Moritz Johner <moolen@users.noreply.github.com>

paul-the-alien[bot] committed:
Merge pull request #937 from external-secrets/chore/getallsecrets-docs

Chore: GetAllSecrets documentation

paul-the-alien[bot] committed:
Merge pull request #913 from rodrmartinez/feature/validate-store-kubernetes-provider

Feature/validateStore for kubernetes provider

paul-the-alien[bot] committed:
Merge pull request #934 from merlindorin/fix/retry-on-error

fix: use controller-runtime internal backoff retry

moolen committed:
feat: implement deletionPolicy (#900)

* feat: implement deletionPolicy

Signed-off-by: Moritz Johner <beller.moritz@googlemail.com>
Co-authored-by: Gustavo Fernandes de Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
adding setup go to release CI

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #939 from external-secrets/chore/update-crane

adding setup go to release CI

gusfcarvalho committed:
update chart version to 0.5.0

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #940 from external-secrets/chore/bump-0.5.0

update chart version to 0.5.0

gusfcarvalho committed:
Refactoring vault provider. Removing RawRequest in favor of Logical

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
GCP: Adds checks to see if a key name exists before trying to load a nested value.

Fixes #941

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #942 from external-secrets/fix/gcp-keys-with-dot

GCP: Adds checks to see if a key name exists before trying to load a nested value

Commit:
feat: gcp getall implementaion

merlindorin committed:
Delegate Vault retries to the controller

gusfcarvalho committed:
Removing Path trimming from hashicorp Vault

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
Updates README.md links to point to latest version.

Fixes #951

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #952 from external-secrets/fix/add-latest-path-to-readme

Updates README.md links to point to latest version.

paul-the-alien[bot] committed:
Merge pull request #950 from external-secrets/fix/disable-path-trimming-from-hashivault

Removing Path trimming from hashicorp Vault

alfredkrohmer committed:
fix: error message for Vault store validation of `Auth.Jwt.KubernetesServiceAccountToken`

alfredkrohmer committed:
fix: only create cert-controller RBAC resources if cert-controller creation itself is enabled

Previously both `certController.create` and `certController.rbac.create` needed to be set to `false` in order to prevent the cert-controller RBAC resources from being created.

paul-the-alien[bot] committed:
Merge pull request #955 from goto-opensource/fix/cert-controller-rbac-conditional

fix: only create cert-controller RBAC resources if cert-controller creation itself is enabled

paul-the-alien[bot] committed:
Merge pull request #954 from goto-opensource/fix/vault-error-message-store-validation-k8s-sa

fix: error message for Vault store validation of `Auth.Jwt.KubernetesServiceAccountToken`

ron1 committed:
Simplify the In-cluster secrets example

Simplify the In-cluster secrets example to work out-of-the-box and ease on-boarding.

Commit:
chore: bump to 0.5.1

paul-the-alien[bot] committed:
Merge pull request #958 from external-secrets/chore/bump0.5.1

Chore/bump0.5.1

Commit:
fix: don't trim path from key

moritzjohner-form3 committed:
docs: add form3 as sponsor

Commit:
fix: check if path is actualy at the start

ron1 committed:
Update docs/provider-kubernetes.md

Co-authored-by: Gustavo Fernandes de Carvalho <gusfcarvalho@gmail.com>

Commit:
docs: starts with filter comment

paul-the-alien[bot] committed:
Merge pull request #949 from merlindorin/feat/improve-retry

Delegate retry to the controller

paul-the-alien[bot] committed:
Merge pull request #962 from external-secrets/feat/add-f3

docs: add form3 as sponsor

paul-the-alien[bot] committed:
Merge pull request #957 from ron1/patch-1

Simplify the In-cluster secrets example

paul-the-alien[bot] committed:
Merge pull request #946 from external-secrets/feat/gcp-getall

feat: gcp getall implementaion

gusfcarvalho committed:
Adding some options for webhook deployment:
 * hostNetwork for webhook pod
 * FailurePolicy for validatingwebhook definition
 * Changed webhook port to a configurable value
 * Defined default value as 9443
Fixes #944

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

gusfcarvalho committed:
Fixes gjson behavior for aws SecretsManager and ParameterStore

Fixes #963

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

Commit:
chore: force interface validation

gusfcarvalho committed:
Adding owner reference to the external secret name, instead of to a generic 'external-secrets' name

Fixes #918

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

knelasevero committed:
Update ADOPTERS.md

paul-the-alien[bot] committed:
Merge pull request #969 from external-secrets/chore/more-adopters

Update ADOPTERS.md

b4ld committed:
add Utility method to validate Network Connection, add Validate method for akeyless,alibaba,gitlab and webhook

knelasevero committed:
Update ADOPTERS.md

paul-the-alien[bot] committed:
Merge pull request #970 from external-secrets/chore/pier-adopter

Update ADOPTERS.md

b4ld committed:
add coonection Close, add default port on Network Validate

b4ld committed:
Merge branch 'external-secrets:main' into feature/validate-gitlab-provider

paul-the-alien[bot] committed:
Merge pull request #967 from external-secrets/chore/force-intf-valid

chore: force interface validation

paul-the-alien[bot] committed:
Merge pull request #966 from external-secrets/fix/aws-keys-with-dot

Fixes gjson behavior for aws SecretsManager and ParameterStore

paul-the-alien[bot] committed:
Merge pull request #968 from external-secrets/fix/creation-policy-merge-behavior

Adding owner reference to the external secret name.

dependabot[bot] committed:
build(deps): bump actions/setup-python from 3.1.0 to 3.1.1

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 3.1.0 to 3.1.1.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v3.1.0...v3.1.1)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump actions/setup-go from 2 to 3

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 2 to 3.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/IBM/secrets-manager-go-sdk

Bumps [github.com/IBM/secrets-manager-go-sdk](https://github.com/IBM/secrets-manager-go-sdk) from 1.0.37 to 1.0.38.
- [Release notes](https://github.com/IBM/secrets-manager-go-sdk/releases)
- [Commits](https://github.com/IBM/secrets-manager-go-sdk/compare/v1.0.37...v1.0.38)

---
updated-dependencies:
- dependency-name: github.com/IBM/secrets-manager-go-sdk
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/aliyun/alibaba-cloud-sdk-go

Bumps [github.com/aliyun/alibaba-cloud-sdk-go](https://github.com/aliyun/alibaba-cloud-sdk-go) from 1.61.1473 to 1.61.1554.
- [Release notes](https://github.com/aliyun/alibaba-cloud-sdk-go/releases)
- [Changelog](https://github.com/aliyun/alibaba-cloud-sdk-go/blob/master/ChangeLog.txt)
- [Commits](https://github.com/aliyun/alibaba-cloud-sdk-go/compare/v1.61.1473...v1.61.1554)

---
updated-dependencies:
- dependency-name: github.com/aliyun/alibaba-cloud-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/lestrrat-go/jwx from 1.2.20 to 1.2.22

Bumps [github.com/lestrrat-go/jwx](https://github.com/lestrrat-go/jwx) from 1.2.20 to 1.2.22.
- [Release notes](https://github.com/lestrrat-go/jwx/releases)
- [Changelog](https://github.com/lestrrat-go/jwx/blob/v1.2.22/Changes)
- [Commits](https://github.com/lestrrat-go/jwx/compare/v1.2.20...v1.2.22)

---
updated-dependencies:
- dependency-name: github.com/lestrrat-go/jwx
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

b4ld committed:
Merge branch 'external-secrets:main' into feature/validate-gitlab-provider

b4ld committed:
fix 4sec timeout on Validate() method to 15sec

paul-the-alien[bot] committed:
Merge pull request #983 from external-secrets/dependabot/go_modules/github.com/lestrrat-go/jwx-1.2.22

build(deps): bump github.com/lestrrat-go/jwx from 1.2.20 to 1.2.22

dependabot[bot] committed:
build(deps): bump github.com/golang-jwt/jwt/v4 from 4.4.0 to 4.4.1

Bumps [github.com/golang-jwt/jwt/v4](https://github.com/golang-jwt/jwt) from 4.4.0 to 4.4.1.
- [Release notes](https://github.com/golang-jwt/jwt/releases)
- [Changelog](https://github.com/golang-jwt/jwt/blob/main/VERSION_HISTORY.md)
- [Commits](https://github.com/golang-jwt/jwt/compare/v4.4.0...v4.4.1)

---
updated-dependencies:
- dependency-name: github.com/golang-jwt/jwt/v4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

paul-the-alien[bot] committed:
Merge pull request #978 from external-secrets/dependabot/github_actions/actions/setup-python-3.1.1

build(deps): bump actions/setup-python from 3.1.0 to 3.1.1

paul-the-alien[bot] committed:
Merge pull request #980 from external-secrets/dependabot/go_modules/github.com/IBM/secrets-manager-go-sdk-1.0.38

build(deps): bump github.com/IBM/secrets-manager-go-sdk from 1.0.37 to 1.0.38

paul-the-alien[bot] committed:
Merge pull request #979 from external-secrets/dependabot/github_actions/actions/setup-go-3

build(deps): bump actions/setup-go from 2 to 3

Commit:
fix: migrate to new api

paul-the-alien[bot] committed:
Merge pull request #982 from external-secrets/dependabot/go_modules/github.com/aliyun/alibaba-cloud-sdk-go-1.61.1554

build(deps): bump github.com/aliyun/alibaba-cloud-sdk-go from 1.61.1473 to 1.61.1554

Commit:
build(deps): bump github.com/aws/aws-sdk-go from 1.41.13 to 1.43.36

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.41.13 to 1.43.36.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Changelog](https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.41.13...v1.43.36)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

paul-the-alien[bot] committed:
Merge pull request #986 from external-secrets/dependabot/go_modules/github.com/golang-jwt/jwt/v4-4.4.1

build(deps): bump github.com/golang-jwt/jwt/v4 from 4.4.0 to 4.4.1

b4ld committed:
Merge branch 'external-secrets:main' into feature/validate-gitlab-provider

paul-the-alien[bot] committed:
Merge pull request #971 from b4ld/feature/validate-gitlab-provider

Network Connection and Validate Methods

Commit:
fix: allow control plane ingress to webhook

moolen committed:
fix: add ssm:DescribeParameters to e2e/managed suite

coreyhinkle committed:
Implement validate for gitlab provider

paul-the-alien[bot] committed:
Merge pull request #960 from bugg123/gitlabProviderValidate

Implement validate for gitlab provider

paul-the-alien[bot] committed:
Merge pull request #981 from external-secrets/dependabot/go_modules/github.com/aws/aws-sdk-go-1.43.36

build(deps): bump github.com/aws/aws-sdk-go from 1.41.13 to 1.43.36

gusfcarvalho committed:
Updated default ports to 10250

Signed-off-by: Gustavo Carvalho <gustavo.carvalho@container-solutions.com>

paul-the-alien[bot] committed:
Merge pull request #965 from external-secrets/fix/webhook-port-and-options

Adding some options for webhook deployment

paul-the-alien[bot] committed:
Merge pull request #959 from external-secrets/chore/refactor-vault

Chore/refactor vault

snarlysodboxer committed:
refactor

snarlysodboxer commented:
I seem to have messed up the rebase on this. The best place to look at the PR that will be submitted is now here: https://github.com/external-secrets/external-secrets/compare/main...Simspace:1Password
